OpenMarket/OpenMarket/Main/MainViewController.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  OpenMarket - MainViewController.swift
+//  Created by yagom. 
+//  Copyright dudu, safari All rights reserved.
+// 
+
+import UIKit
+
+private enum Section {
+    case main
+}
+
+private enum Constant {
+    static let requestItemCount = 20
+}
+
+final class MainViewController: UIViewController {
+    private typealias DataSource = UICollectionViewDiffableDataSource<Section, Product>
+    private typealias Snapshot = NSDiffableDataSourceSnapshot<Section, Product>
+    
+    private lazy var mainView = MainView(frame: self.view.bounds)
+    private lazy var dataSource = makeDataSource()
+    private lazy var snapshot = makeSnapshot()
+    
+    private var pageNumber = 1
+    
+    private var networkManager = NetworkManager<ProductList>(imageCache: CacheManager())
+    
+    private lazy var segmentControl: CellSegmentControl = {
+        let segmentControl = CellSegmentControl(items: ["LIST", "GRID"])
+        segmentControl.addTarget(self, action: #selector(segmentValueDidChanged(segmentedControl:)), for: .valueChanged)
+        return segmentControl
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureView()
+        configureNavigationBar()
+        configureRefreshControl()
+        requestData(pageNumber: pageNumber)
+        
+    }
+    
+    private func configureView() {
+        mainView.backgroundColor = .systemBackground
+        self.view = mainView
+        configureCollectionView()
+    }
+    
+    private func configureNavigationBar() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(addButtonDidTapped))
+        navigationItem.titleView = segmentControl
+    }
+    
+    private func configureCollectionView() {
+        mainView.collectionView.register(ProductGridCell.self, forCellWithReuseIdentifier: ProductGridCell.identifier)
+        mainView.collectionView.register(ProductListCell.self, forCellWithReuseIdentifier: ProductListCell.identifier)
+        mainView.collectionView.prefetchDataSource = self
+    }
+    
+    private func configureRefreshControl() {
+        mainView.collectionView.refreshControl = UIRefreshControl()
+        mainView.collectionView.refreshControl?.addTarget(self, action: #selector(handleRefreshControl), for: .valueChanged)
+    }
+}
+
+// MARK: - Action Method
+
+extension MainViewController {
+    @objc private func addButtonDidTapped() {}
```

Comment:
빈 메서드일 경우에는 주석으로 표시해주는 것이 좋습니다.
```suggestion
    @objc private func addButtonDidTapped() {
        // empty
    }
```